app/services/order_cycle_form.rb

```diff
@@ -61,14 +61,29 @@ def apply_exchange_changes
   def attach_selected_distributor_payment_methods
     return if @selected_distributor_payment_method_ids.nil?
 
-    order_cycle.selected_distributor_payment_method_ids = selected_distributor_payment_method_ids
+    if distributor_only?
+      payment_method_ids = order_cycle.selected_distributor_payment_method_ids
+      payment_method_ids -= user_distributor_payment_method_ids
+      payment_method_ids += user_only_selected_distributor_payment_method_ids
+      order_cycle.selected_distributor_payment_method_ids = payment_method_ids
+    else
+      order_cycle.selected_distributor_payment_method_ids = selected_distributor_payment_method_ids
+    end
     order_cycle.save!
   end
 
   def attach_selected_distributor_shipping_methods
     return if @selected_distributor_shipping_method_ids.nil?
 
-    order_cycle.selected_distributor_shipping_method_ids = selected_distributor_shipping_method_ids
+    if distributor_only?
+      shipping_method_ids = order_cycle.selected_distributor_shipping_method_ids
+      shipping_method_ids -= user_distributor_shipping_method_ids
+      shipping_method_ids += user_only_selected_distributor_shipping_method_ids
+      order_cycle.selected_distributor_shipping_method_ids = shipping_method_ids
```

Comment:
To understand this requires a bit more context, so I suggest a comment such as:

```suggestion
      # A distributor can only update methods associated with their own
      # enterprise, so we load all previously selected methods, and replace
      # only the distributor's methods with their selection (not touching other
      # distributor's methods).
      shipping_method_ids = order_cycle.selected_distributor_shipping_method_ids
      shipping_method_ids -= user_distributor_shipping_method_ids
      shipping_method_ids += user_only_selected_distributor_shipping_method_ids
      order_cycle.selected_distributor_shipping_method_ids = shipping_method_ids
```

Reply:
This is better